jq.c

```diff
@@ -277,6 +289,96 @@ jdbc_detach_jvm()
 	(*jvm)->DetachCurrentThread(jvm);
 }
 
+static void jdbc_get_jni_env(void)
+{
+	int JVMEnvStat;
+
+	ereport(DEBUG3, (errmsg("In In jdbc_get_jni_env")));
+
+	JVMEnvStat = (*jvm)->GetEnv(jvm, (void **) &Jenv, JNI_VERSION);
+	if (JVMEnvStat == JNI_EDETACHED)
+	{
+		ereport(DEBUG3, (errmsg("JVMEnvStat: JNI_EDETACHED; the current thread is not attached to the VM")));
+		jdbc_attach_jvm();
+	}
+	else if (JVMEnvStat == JNI_OK)
+	{
+		ereport(DEBUG3, (errmsg("JVMEnvStat: JNI_OK")));
+	}
+	else if (JVMEnvStat == JNI_EVERSION)
+	{
+		ereport(ERROR, (errmsg("JVMEnvStat: JNI_EVERSION; the specified version is not supported")));
+	}
+}
+
+static void jdbc_add_classpath_to_system_class_loader(char *classpath)
+{
+	int url_classpath_len;
+	char *url_classpath;
+	jclass ClassLoader_class;
+	jmethodID ClassLoader_getSystemClassLoader;
+	jobject system_class_loader;
+	jclass URLClassLoader_class;
+	jmethodID URLClassLoader_addURL;
+	jclass URL_class;
+	jmethodID URL_constructor;
+	jobject url;
+
+	ereport(DEBUG3, errmsg("In jdbc_add_classpath_to_system_class_loader"));
+
+	url_classpath_len = 5 + strlen(classpath) + 2; /* "file:" + classpath + '/\0' */
+	url_classpath = (char *)palloc0(url_classpath_len);
+	snprintf(url_classpath, url_classpath_len, "file:%s/", classpath);
+
+
+	ClassLoader_class = (*Jenv)->FindClass(Jenv, "java/lang/ClassLoader");
+	if (ClassLoader_class == NULL) {
+		ereport(ERROR, errmsg("java/lang/ClassLoader is not found"));
+	}
+
+	ClassLoader_getSystemClassLoader = (*Jenv)->GetStaticMethodID(Jenv, ClassLoader_class,
+															   "getSystemClassLoader", "()Ljava/lang/ClassLoader;");
```

Comment:
Sorry, I am not familiar with C formatting but is the format correct here?
Two other method calls below in this method also have the last parameter set on the next line.

Reply:
These codes are syntactically correct in C. The only reason why I separated these into multiple lines is to avoid too long lines. Such formatting also exists in the current code, e.g. [here](https://github.com/scalar-labs/jdbc_fdw/blob/ce1a3c3db82618f67e2693995558b469874f3665/jq.c#L988-L989).

Reply:
I see, thank you.